v1/lists.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const P = require('bluebird');
+const preq = require('preq');
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const spec = HyperSwitch.utils.loadSpec(`${__dirname}/lists.yaml`);
+
+class ReadingLists {
+    /**
+     * @param {!Object} options RESTBase options object.
+     *   Apart from standard options, the following are recognized:
+     *   - summaryConcurrency: the max number of parallel requests to summary endpoint (default: 10)
+     */
+    constructor(options) {
+        this.options = Object.assign({}, options, {
+            summaryConcurrency: 10,
+        });
+    }
+
+    /**
+     * Transform the continuation data into a string so it is easier for clients to deal with.
+     * @param {!Object|undefined} continuation Continuation object returned by the MediaWiki API.
+     * @return {!String|undefined} Continuation string.
+     */
+    flattenContinuation(continuation) {
+        return JSON.stringify(continuation);
+    }
+
+    /**
+     * Inverse of flattenContinuation.
+     * @param {!String|undefined} continuation Continuation string returned by flattenContinuation()
+     * @return {!Object} Continuation object.
+     */
+    unflattenContinuation(continuation) {
+        const sanitizedContinuation = {};
+        if (typeof continuation === 'string') {
+            try {
+                continuation = JSON.parse(continuation);
+            } catch (e) {
+                this.options.log('error/unflatten', {
+                    msg: e.stack,
```

Comment:
The stack is not a great message for a log entry, something descriptive would be much better.